reference-architecture/gce-cli/gcloud.sh

```diff
@@ -360,24 +383,38 @@ if ! gcloud --project "$GCLOUD_PROJECT" compute project-info describe | grep -q
 fi
 
 # Create pre-registered image based on the uploaded image
-if ! gcloud --project "$GCLOUD_PROJECT" compute images describe "$REGISTERED_IMAGE" &>/dev/null; then
-    gcloud --project "$GCLOUD_PROJECT" compute instances create "$TEMP_INSTANCE" --zone "$GCLOUD_ZONE" --machine-type "n1-standard-1" --network "$OCP_NETWORK" --image "$RHEL_IMAGE_GCE" --boot-disk-size "10" --no-boot-disk-auto-delete --boot-disk-type "pd-ssd" --tags "ssh-external"
+if [ -n "$REGISTERED_IMAGE" ] && gcloud --project "$GCLOUD_PROJECT" compute images describe "$REGISTERED_IMAGE" &>/dev/null; then
+    echo "Image '${REGISTERED_IMAGE}' already exists"
+elif [ -n "$REGISTERED_IMAGE" ]; then
+    if [ -z "$PARENT_IMAGE_SELECTOR" ]; then
+        PARENT_IMAGE_SELECTOR="--image $RHEL_IMAGE_GCE"
+    fi
+
+    gcloud --project "$GCLOUD_PROJECT" compute instances create "$TEMP_INSTANCE" --zone "$GCLOUD_ZONE" --machine-type "n1-standard-1" --network "$OCP_NETWORK" $PARENT_IMAGE_SELECTOR --boot-disk-size "10" --no-boot-disk-auto-delete --boot-disk-type "pd-ssd" --tags "ssh-external"
     until gcloud -q --project "$GCLOUD_PROJECT" compute ssh "cloud-user@${TEMP_INSTANCE}" --zone "$GCLOUD_ZONE" --command "echo" &>/dev/null; do
         echo "Waiting for '${TEMP_INSTANCE}' to come up..."
         sleep 5
     done
     if ! gcloud -q --project "$GCLOUD_PROJECT" compute ssh "cloud-user@${TEMP_INSTANCE}" --zone "$GCLOUD_ZONE" --ssh-flag="-t" --command "sudo bash -euc '
-        subscription-manager register --username=${RH_USERNAME} --password=\"${RH_PASSWORD}\";
-        subscription-manager attach --pool=${RH_POOL_ID};
-        subscription-manager repos --disable=\"*\";
-        subscription-manager repos \
-            --enable=\"rhel-7-server-rpms\" \
-            --enable=\"rhel-7-server-extras-rpms\" \
-            --enable=\"rhel-7-server-ose-${OCP_VERSION}-rpms\";
+        if which subscription-manager; then
+            subscription-manager register --username=${RH_USERNAME} --password=\"${RH_PASSWORD}\";
+            subscription-manager attach --pool=${RH_POOL_ID};
+            subscription-manager repos --disable=\"*\";
+            subscription-manager repos \
+                --enable=\"rhel-7-server-rpms\" \
+                --enable=\"rhel-7-server-extras-rpms\" \
+                --enable=\"rhel-7-server-ose-${OCP_VERSION}-rpms\";
```

Comment:
@bdurrow woudl probably be better to key off of the deployment_type if possible.